boards/arm/stm32f746g_disco/doc/stm32f746g_disco.rst

```diff
@@ -0,0 +1,200 @@
+
+- STM32F746NGH6 in BGA216 package
+- ARM |reg| 32-bit Cortex |reg| -M7 CPU with FPU
+- 216 MHz max CPU frequency
+- VDD from 1.8 V to 3.6 V
+- 2 MB Flash
+- 384+4 KB SRAM including 64-Kbyte of core coupled memory
+- GPIO with external interrupt capability
+- LCD parallel interface, 8080/6800 modes
+- LCD TFT controller supporting up to XGA resolution
+- MIPI |reg|  DSI host controller supporting up to 720p 30Hz resolution
+- 3x12-bit ADC with 24 channels
+- 2x12-bit D/A converters
+- RTC
+- Advanced-control Timer
+- General Purpose Timers (17)
+- Watchdog Timers (2)
+- USART/UART (8)
+- I2C (3)
+- SPI (6)
+- 1xSAI (serial audio interface)
+- SDIO
+- 2xCAN
+- USB 2.0 OTG FS with on-chip PHY
+- USB 2.0 OTG HS/FS with dedicated DMA, on-chip full-speed PHY and ULPI
+- 10/100 Ethernet MAC with dedicated DMA
+- 8- to 14-bit parallel camera
+- CRC calculation unit
+- True random number generator
+- DMA Controller
+
+More information about STM32F746NGH6 can be found here:
+       - `STM32F746NGH6 on www.st.com`_
+       - `STM32F74xxx reference manual`_
+
+Supported Features
+==================
+
+The Zephyr stm32f746g_disco board configuration supports the following hardware features:
+
++-----------+------------+-------------------------------------+
+| Interface | Controller | Driver/Component                    |
++===========+============+=====================================+
+| NVIC      | on-chip    | nested vector interrupt controller  |
++-----------+------------+-------------------------------------+
+| UART      | on-chip    | serial port-polling;                |
+|           |            | serial port-interrupt               |
++-----------+------------+-------------------------------------+
+| PINMUX    | on-chip    | pinmux                              |
++-----------+------------+-------------------------------------+
+| GPIO      | on-chip    | gpio                                |
++-----------+------------+-------------------------------------+
+
+Other hardware features are not yet supported on Zephyr porting.
+
+The default configuration can be found in the defconfig file:
+
+	``boards/arm/stm32f746g_disco/stm32f746g_disco_defconfig``
+
+
+Pin Mapping
+===========
+
+STM32F746G Discovery kit has 9 GPIO controllers. These controllers are responsible for pin muxing,
+input/output, pull-up, etc.
+
+For mode details please refer to `32F746G-DISCO board User Manual`_.
+
+Default Zephyr Peripheral Mapping:
+----------------------------------
+- UART_1_TX : PA9
+- UART_1_RX : PB7
+- UART_3_TX : PC6
+- UART_3_RX : PC7
+- LD1 : PI1
+
+System Clock
+============
+
+STM32F746G System Clock could be driven by internal or external oscillator,
+as well as main PLL clock. By default System clock is driven by PLL clock at 216MHz,
+driven by 25MHz high speed external clock.
+
+Serial Port
+===========
+
+The STM32F746G Discovery kit has up to 8 UARTs. The Zephyr console output is assigned to UART1.
```

Comment:
I suspect the kit has a fixed number of UARTs, so it's better to say:

    The STM32F746G Discovery kit has 8 UARTs.

Reply:
@dbkinder, 
I am not sure that all 8 UARTs are available. Many ports are occupied by LCD display with touchscreen and other peripherals.